dos/src/main.rs

```diff
@@ -23,6 +29,53 @@ fn get_repair_contact(nodes: &[ContactInfo]) -> ContactInfo {
     contact
 }
 
+fn test_multisig_tx(
+    transaction_params: &TransactionParams,
+    rpc_client: &Option<RpcClient>,
+) -> Transaction {
+    let kpvals: Vec<Keypair> = (0..transaction_params.num_sign)
+        .map(|_| Keypair::new())
+        .collect();
+    let keypairs: Vec<&Keypair> = kpvals.iter().collect();
+
+    let blockhash = if transaction_params.valid_block_hash {
+        rpc_client.as_ref().unwrap().get_latest_blockhash().unwrap()
```

Comment:
Retrieving a new blockhash on every single new transaction may not be that fast. Probably want to cache it and fetch every few seconds.